### PR TITLE
feat(report): Include dependencies into scan result and cyclondex for supply chain security on Integration with GitHub Security Alerts

### PR DESCRIPTION
**Pull request by kl-sinclair**
# What did you implement:

- When you integrate with GitHub Security Alerts, Vuls did NOT take non-vulnerable packages in SBOM
- From now on SBOM reporting, it fetches Dependency graph together with GitHub Security Alerts
- And that becomes components and dependencies
- Also, Package URL of vulnerable packages are linked to vulnerabilities in affects

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

```
$ vuls report -to-localfile -format-cyclonedx-json
or 
$ vuls report -to-localfile -format-cyclonedx-xml
```

# Checklist:
You don't have to satisfy all of the following.

- [ ] Write tests
- [ ] Write documentation
- [x] Check that there aren't other open pull requests for the same issue/feature
- [x] Format your source code by `make fmt`
- [x] Pass the test by `make test`
- [x] Provide verification config / commands
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES

# Reference

https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-the-dependency-graph

### CODE REVIEW

File: detector/detector.go [hunk 1]
```diff
@@ -303,6 +303,10 @@ func DetectGitHubCves(r *models.ScanResult, githubConfs map[string]config.GitHub
 		}
 		logging.Log.Infof("%s: %d CVEs detected with GHSA %s/%s",
 			r.FormatServerName(), n, owner, repo)
+
+		if err = DetectGitHubDependencyGraph(r, owner, repo, setting.Token); err != nil {
+			return xerrors.Errorf("Filed to access GitHub Dependency graph: %w", err)
```
Comment (MaineK00n) [inline detector/detector.go:308]:
```suggestion
			return xerrors.Errorf("Failed to access GitHub Dependency graph: %w", err)
```

File: detector/github.go [hunk 2]
```diff
@@ -199,3 +203,146 @@ type SecurityAlerts struct {
...
+			return err
+		}
+
+		if graph.Data.Repository.URL == "" {
+			return errof.New(errof.ErrFailedToAccessGithubAPI,
+				fmt.Sprintf("Failed to access to GitHub API. Response: %s", string(body)))
+		}
+
+		dependenciesAfter = ""
+		for _, m := range graph.Data.Repository.DependencyGraphManifests.Edges {
```
Comment (kotakanbe) [inline detector/github.go:260]: Plz create functions for maintainability

File: detector/github.go [hunk 3]
```diff
@@ -199,3 +203,146 @@ type SecurityAlerts struct {
...
+	//TODO Proxy
+	httpClient := oauth2.NewClient(context.Background(), src)
+
+	const jsonfmt = `{"query":
+	"query { repository(owner:\"%s\", name:\"%s\") { url dependencyGraphManifests(first: %d, withDependencies: true%s) { pageInfo { endCursor hasNextPage } edges { node { blobPath filename repository { url } parseable exceedsMaxSize dependenciesCount dependencies%s { pageInfo { endCursor hasNextPage } edges { node { packageName packageManager repository { url } requirements hasDependencies } } } } } } } }"}`
+	after := ""
+	dependenciesAfter := ""
+
+	r.GitHubPackages = models.DependencyGraphManifests{}
+	for {
```
Comment (kotakanbe) [inline detector/github.go:221]: I think the recursive function is easier to understand

File: detector/github.go (hunk 2 above)
Comment (kl-sinclair) [reply detector/github.go]: done

File: detector/github.go (hunk 3 above)
Comment (kl-sinclair) [reply detector/github.go]: done

**Comment by kl-sinclair** [comment]
Sample SBOM output
- https://github.com/kl-sinclair/isucon6
  - https://gist.github.com/kl-sinclair/eaaa5839c37da07d68c130a409395999
- https://github.com/kl-sinclair/vuls
  - https://gist.github.com/kl-sinclair/11a387630f3d7c5f17e01f7924290175

**Review by kotakanbe — APPROVED** [review]
LGTM

**Pull request merged**